sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-import.service.ts

```diff
@@ -0,0 +1,114 @@
+import { Injectable } from "@nestjs/common";
+import {
+  ApplicationBulkWithdrawalHeader,
+  ApplicationWithdrawalTextValidationResult,
+} from "./application-bulk-withdrawal-import-text.models";
+import { In, Repository } from "typeorm";
+import { Application } from "@sims/sims-db";
+import {
+  ApplicationBulkWithdrawalValidationModel,
+  ApplicationData,
+} from "./application-bulk-withdrawal-validation.models";
+import { InjectRepository } from "@nestjs/typeorm";
+
+type ApplicationDataMap = Record<string, ApplicationData>;
+
+/**
+ * Handles the application withdrawal bulk insert preparation.
+ */
+@Injectable()
+export class ApplicationBulkWithdrawalImportService {
+  constructor(
+    @InjectRepository(Application)
+    private readonly applicationRepo: Repository<Application>,
+  ) {}
+
+  /**
+   * Generates the application bulk withdrawal validation model.
+   * @param textValidations text validation models to be converted to the validation models.
+   * @param textHeader text header model.
+   * @param institutionId institution id.
+   * @returns application bulk withdrawal models to be validated and persisted.
+   */
+  async generateApplicationBulkWithdrawalValidationModels(
+    textValidations: ApplicationWithdrawalTextValidationResult[],
+    textHeader: ApplicationBulkWithdrawalHeader,
+    institutionId: number,
+  ): Promise<ApplicationBulkWithdrawalValidationModel[]> {
+    const applicationDataMap: ApplicationDataMap = {};
+    const applicationNumbers = textValidations.map(
+      (textValidation) => textValidation.textModel.applicationNumber,
+    );
+    const applicationValidationData = await this.getApplicationValidationData(
+      applicationNumbers,
+      institutionId,
+    );
+    applicationValidationData.forEach((record) => {
+      applicationDataMap[record.applicationNumber] = {
+        applicationStatus: record.applicationStatus,
+        isArchived: record.isArchived,
+        sin: record.student.sinValidation.sin,
+        locationId: record.location.id,
+        locationCode: record.location.institutionCode,
+        hasPreviouslyBeenWithdrawn: !!record.studentScholasticStandings.length,
```

Comment:
This is the property that I am not following and that also supports the comment: https://github.com/bcgov/SIMS/pull/2465#discussion_r1379375219
If the property is named `hasPreviouslyBeenWithdrawn` but is actually checking for any "Scholastic Standing" it does not seem right.

Depending on what the business answered, at least the property must be changed to `hasScholasticStanding` and the message should also be adjusted from "The application is already withdrawn with the date..." to "The application already has a Scholastic Standing".

Reply:
Keeping it as is after the discussion with business [@HRAGANBC]

Reply:
It is ok to keep it now since the SQL query is changed and `studentScholasticStandings` will contain only the withdrawals.